internal/binary/binary.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package binary provides support for binary operations on files.
+package binary
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+)
+
+// ChecksMagic checks the magic bytes from the provided reader.
```

Comment:
typo: CheckMagic not ChecksMagic

// CheckMagic reads len(want) bytes from r and returns an error on mismatch.

Typing the comment makes me think this function should just be called "ExpectBytes" or something since it is generally useful not just for checking magic.

Then at the call sites:

if err := binary.ExpectBytes(r, []byte(bamMagic)); err != nil {
  return fmt.Errorf("reading magic: %v", err)
}

sort of thing

Reply:
done